packages/valory/connections/ledger/tests/test_contract_dispatcher.py

```diff
@@ -0,0 +1,280 @@
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the tests of the ledger connection module."""
+
+from unittest import mock
+from unittest.mock import ANY, MagicMock, Mock, patch
+
+import pytest
+from aea_ledger_ethereum import EthereumCrypto
+
+from aea.common import Address
+from aea.crypto.ledger_apis import ETHEREUM_DEFAULT_ADDRESS
+from aea.crypto.registries import ledger_apis_registry
+from aea.exceptions import AEAException
+from aea.helpers.async_utils import AsyncState
+from aea.multiplexer import MultiplexerStatus
+from aea.protocols.base import Message
+from aea.protocols.dialogue.base import Dialogue
+
+from packages.valory.connections.ledger.contract_dispatcher import (
+    ContractApiRequestDispatcher,
+)
+
+# pylint: skip-file
+from packages.valory.protocols.contract_api import ContractApiMessage
+from packages.valory.protocols.contract_api.dialogues import ContractApiDialogue
+from packages.valory.protocols.contract_api.dialogues import (
+    ContractApiDialogues as BaseContractApiDialogues,
+)
+
+
+SOME_SKILL_ID = "some/skill:0.1.0"
+NON_BLOCKING_TIME = 1
+BLOCKING_TIME = 100
+TOLERANCE = 1
+WAIT_TIME_AMONG_TASKS = 0.1
+
+
+class ContractApiDialogues(BaseContractApiDialogues):
+    """This class keeps track of all contract_api dialogues."""
+
+    def __init__(self, self_address: str) -> None:
+        """
+        Initialize dialogues.
+
+        :param self_address: the address of the entity for whom dialogues are maintained
+        """
+
+        def role_from_first_message(  # pylint: disable=unused-argument
+            message: Message, receiver_address: Address
+        ) -> Dialogue.Role:
+            """Infer the role of the agent from an incoming/outgoing first message
+
+            :param message: an incoming/outgoing first message
+            :param receiver_address: the address of the receiving agent
+            :return: The role of the agent
+            """
+            return ContractApiDialogue.Role.AGENT
+
+        BaseContractApiDialogues.__init__(
+            self,
+            self_address=self_address,
+            role_from_first_message=role_from_first_message,
+        )
+
+
+def test_validate_and_call_callable() -> None:
+    """Tests a default method call through ContractApiRequestDispatcher."""
+
+    dummy_address = "0x0000000000000000000000000000000000000000"
+    contract_dispatcher = ContractApiRequestDispatcher(
+        connection_id=Mock(), connection_state=AsyncState()
+    )
+    contract = Mock()
+
+    def callable(ledger_api=None, contract_address=None, _addr=None):  # type: ignore
+        return 0
```

Comment:
overwrite of built-in function `callable`, please rename

Reply:
done